File: 2024/Day_11/solution.py
def process_stones(stones, blinks):
    for _ in range(blinks):
        new_stones = []
        for stone in stones:
            if stone == 0:
                new_stones.append(1)
            elif len(str(stone)) % 2 == 0:
                str_stone = str(stone)
                mid = len(str_stone) // 2
                left = int(str_stone[:mid])
                right = int(str_stone[mid:])
                new_stones.extend([left, right])
            else:
                new_stones.append(stone * 2024)
        stones = new_stones
    return stones

# Example usage
initial_sample_stones = [125, 17]
initial_stones = [337, 42493, 1891760, 351136, 2, 6932, 73, 0]
blinks_part1 = 25
blinks_part2 = 75

result_stones_part1 = process_stones(initial_sample_stones, blinks_part1)
print(f"Number of stones after {blinks_part1} blinks: {len(result_stones_part1)}")

## Long prosessing times to get 75 blinks!
## From chatGBT: Processing for blinks 75, Attempting this calculation would likely exceed memory limits or result in excessive computation time, making it impractical to compute.
result_stones_part2 = process_stones(initial_sample_stones, blinks_part2)
print(f"Number of stones after {blinks_part2} blinks: {len(result_stones_part2)}")

File: 2024/Day_11/solution.js
const fs = require('fs');

// Read input from the specified file
const filename = 'input-sample.txt';
const inputs = fs.readFileSync(filename, 'utf8').trim().split('\n');

// Function to process stones based on the rules
function processStones(stones, blinks) {
    for (let i = 0; i < blinks; i++) {
        const newStones = [];
        for (const stone of stones) {
            if (stone === 0) {
                newStones.push(1);
            } else if (stone.toString().length % 2 === 0) {
                const strStone = stone.toString();
                const mid = strStone.length / 2;
                const left = parseInt(strStone.slice(0, mid), 10);
                const right = parseInt(strStone.slice(mid), 10);
                newStones.push(left, right);
            } else {
                newStones.push(stone * 2024);
            }
        }
        stones = newStones;
    }
    return stones;
}

// Parse the initial stones from the input
const initialStones = inputs[0].split(' ').map(Number);

// Define the number of blinks for Part 1 and Part 2
const blinksPart1 = 25;
const blinksPart2 = 75;

// Process the stones for Part 1
const resultStonesPart1 = processStones([...initialStones], blinksPart1);
console.log(`Number of stones after ${blinksPart1} blinks: ${resultStonesPart1.length}`);

// Process the stones for Part 2
// Due to memory issue this solution not working, tried in python also
const resultStonesPart2 = processStones([...initialStones], blinksPart2);
console.log(`Number of stones after ${blinksPart2} blinks: ${resultStonesPart2.length}`);
